packages/language-service/ivy/test/references_spec.ts

```diff
@@ -0,0 +1,674 @@
+    const {text, cursor} = extractCursorInfo(`
+          import {Component} from '@angular/core';
+
+          @Component({templateUrl: './app.html'})
+          export class AppCmp {
+            myP¦rop!: string;
+          }`);
+    const appFile = {name: _('/app.ts'), contents: text};
+    const templateFile = {name: _('/app.html'), contents: '{{myProp}}'};
+    createModuleWithDeclarations([appFile], [templateFile]);
+    const refs = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    expect(refs.length).toBe(2);
+    assertFileNames(refs, ['app.html', 'app.ts']);
+    assertTextSpans(refs, ['myProp']);
+  });
+
+  it('gets component member references from template', () => {
+    const appFile = {
+      name: _('/app.ts'),
+      contents: `
+          import {Component} from '@angular/core';
+
+          @Component({templateUrl: './app.html'})
+          export class AppCmp {
+            myProp = '';
+          }`
+    };
+    const {text, cursor} = extractCursorInfo('{{myP¦rop}}');
+    const templateFile = {name: _('/app.html'), contents: text};
+    createModuleWithDeclarations([appFile], [templateFile]);
+    const refs = getReferencesAtPosition(_('/app.html'), cursor)!;
+    // Angular LS returns the template ref _and_ component member because TSLS doesn't know how to
+    // find it from the template.
+    expect(refs.length).toBe(2);
+    assertFileNames(refs, ['app.html', 'app.ts']);
+    assertTextSpans(refs, ['myProp']);
+  });
+
+  it('should work for method calls', () => {
+    const {text, cursor} = extractCursorInfo(`
+          import {Component} from '@angular/core';
+
+          @Component({template: '<div (click)="set¦Title(2)"></div>'})
+          export class AppCmp {
+            setTitle(s: number) {}
+          }`);
+    const appFile = {name: _('/app.ts'), contents: text};
+    createModuleWithDeclarations([appFile]);
+    const refs = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    expect(refs.length).toBe(2);
+
+    assertFileNames(refs, ['app.ts']);
+    assertTextSpans(refs, ['setTitle']);
+  });
+
+  it('should work for method call arguments', () => {
+    const {text, cursor} = extractCursorInfo(`
+          import {Component} from '@angular/core';
+
+          @Component({template: '<div (click)="setTitle(ti¦tle)"></div>'})
+          export class AppCmp {
+            title = '';
+            setTitle(s: string) {}
+          }`);
+    const appFile = {name: _('/app.ts'), contents: text};
+    createModuleWithDeclarations([appFile]);
+    const refs = getReferencesAtPosition(_('/app.ts'), cursor)!;
+    expect(refs.length).toBe(2);
+
+    assertTextSpans(refs, ['title']);
+  });
+
+  it('should work for property writes', () => {
+    const appFile = {
+      name: _('/app.ts'),
+      contents: `
+          import {Component} from '@angular/core';
+
+          @Component({templateUrl: './app.html' })
+          export class AppCmp {
+            title = '';
+          }`
```

Comment:
very nit, here and other places: 
```suggestion
          }`,
```